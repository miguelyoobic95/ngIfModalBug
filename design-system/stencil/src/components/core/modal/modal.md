---
name: Modal
category: Core Components
---

Modals can be customized with multiple properties. Properties conform to the IModalEntry interface below:

```ts
export interface IModalEntry {
    heading?: string;
    headingIcon?: string;
    content?: HTMLElement;
    hasHeader?: boolean;
    hasFooter?: boolean;
    footerText?: string;
    primaryButtonText?: string;
    secondaryButtonText?: string;
    cssClass?: string;
    animationName?: string;
    animationProp?: any;
    primaryFn?: Function;
    withYooCtrl?: boolean;
    scrollEnabled?: boolean;
}
```

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`heading`|string|   |the heading of the modal, will only be displayed if `hasHeader` property is `true`|
|`headingIcon`|string| |the icon to display to the left of the heading|
|`content`|HTMLElement| |the content inside the body of the modal|
|`hasHeader`|boolean|`true`|defines if the modal has a header|
|`hasFooter`|boolean|`false`|defines if the modal has a footer|
|`footerText`|string||the text inside the footer|
|`primaryButtonText`|string|`Confirm`  |the text inside the primary button, will only be displayed if `hasFooter` property is `true`|
|`secondaryButtonText`|string|`Cancel`|the text inside the secondary button, will only be displayed if `hasFooter` property is `true`|
|`cssClass`|string| |space separated classes to add to the modal|
|`animationName`|string||adds an animation to the modal, see animation docs for options|
|`animationProp`|any|||
|`primaryFn`|Function||function triggered on the modal primary button click|
|`withYooCtrl`|boolean|specify whether the `yoo-modal-controller` is being used|
|`scrollEnabled`|boolean|false|specify whether the `yoo-slim-scroll` is being used|

### Basic

```yoo-modal-basic.html
    <div style="height: 500px; width: 500px;">
        <yoo-modal heading="Basic Modal" with-yoo-ctrl="true">
            <div >Test content</div>        
        </yoo-modal>
    </div>
```

### With Footer

```yoo-modal-footer.html
    <div style="height: 500px; width: 500px;">
        <yoo-modal class="accent" with-yoo-ctrl="true" headingIcon="yo-fire" heading="Modal with footer" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
            <div >Test content</div>
        </yoo-modal>
    </div>
```

### Without Header

```yoo-modal-no-header.html

    <yoo-modal has-footer="true" with-yoo-ctrl="true" has-header="false">
        <div >Test content</div>
    </yoo-modal>
```

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|yoo-modal accent|
|Dark|`dark`|yoo-modal dark|
|Success|`success`|yoo-modal success|
|Danger|`danger`|yoo-modal danger|
|Info|`info`|yoo-modal info|
|Warning|`warning`|yoo-modal warning|
|Fullscreen|`fullscreen`|yoo-modal fullscreen|
|Drawer|`drawer`|yoo-modal drawer|

```yoo-modal-danger.html
    <yoo-modal class="danger" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal danger" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-success.html
    <yoo-modal class="success" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal success" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-info.html
    <yoo-modal class="info" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal info" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-warning.html
    <yoo-modal class="warning" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal warning" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-dark.html
    <yoo-modal class="dark" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal dark" has-footer="true" primary-button-text="Confirm" secondary-button-text="Dismiss">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-drawer.html
    <yoo-modal class=" drawer accent" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal dark" has-footer="true" primary-button-text="Confirm" secondary-button-text="Dismiss">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-fullscreen.html
    <yoo-modal class="success fullscreen" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Fullscreen Success" has-footer="true" primary-button-text="Confirm" secondary-button-text="Dismiss">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-menu.html
    <yoo-modal class="accent menu" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Menu Accent" has-footer="true" primary-button-text="Confirm" secondary-button-text="Dismiss">
        <div >Test content</div>
    </yoo-modal>
```

## Events

|Attr|Description|
|---|---|
|`modalPrimaryButtonClicked`|called when the primary button is clicked|
|`modalClosed`|called when the modal is closed|