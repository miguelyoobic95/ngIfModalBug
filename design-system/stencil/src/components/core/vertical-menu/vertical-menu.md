---
name: Vertical Menu
category: Core Components
---

Vertical Menu is intended to be used as the main menu of an application. It displays of the left.

The following interfaces are used:

```tsx 
export interface IVerticalMenuEntry {
    menuRows: IVerticalMenuRow[];
}

export interface IVerticalMenuRow {
    item?: IVerticalMenuItem;
    subItems?: IVerticalMenuItem[];
}

export interface IVerticalMenuItem {
    text?: string;
    imgSrc?: string;
    icon?: string;
    emittedEvent?: string;
    isActive?: boolean;
    anchor?: string;
}
```

```yoo-vertical-menu.html
    <yoo-vertical-menu></yoo-vertical-menu>
```
```yoo-vertical-menu.js 
  var menu = document.querySelector('yoo-vertical-menu');
  var verticalText = { text: "Amazing Component"};
  var verticalTextActive = { text: "Text", isActive: true };
  var verticalText2 = { text: "Core Component", imgSrc: "./assets/empty-states/calendar.svg" };
  var verticalText3 = { text: "Form Component", imgSrc: "./assets/empty-states/channel.svg", isActive: true };
  var verticalRow = { item: verticalText};
  var verticalRowActive = { item: verticalTextActive};
  var verticalRow2 = { item: verticalText2, subItems: [verticalText, verticalText3] };
  var verticalRow3 = { item: verticalText, subItems: [verticalText, verticalText, verticalText] };
  menu.entry = { menuRows: [verticalRowActive, verticalRow, verticalRow2, verticalRow, verticalRow3] };
```

```yoo-vertical-menu-side.html
 <div style="width:50%">
    <yoo-vertical-menu></yoo-vertical-menu>
 </div>
```
```yoo-vertical-menu-side.js 
  var menu = document.querySelector('yoo-vertical-menu');
  var verticalText = { text: "Amazing Component"};
  var verticalTextActive = { text: "Text", isActive: true };
  var verticalText2 = { text: "Core Component", imgSrc: "./assets/empty-states/calendar.svg" };
  var verticalText3 = { text: "Form Component", imgSrc: "./assets/empty-states/channel.svg", isActive: true };
  var verticalRow = { item: verticalText};
  var verticalRowActive = { item: verticalTextActive};
  var verticalRow2 = { item: verticalText2, subItems: [verticalText, verticalText3] };
  var verticalRow3 = { item: verticalText, subItems: [verticalText, verticalText, verticalText] };
  menu.entry = { menuRows: [verticalRowActive, verticalRow, verticalRow2, verticalRow, verticalRow3] };
```

```yoo-vertical-menu-modal.html
    <yoo-vertical-menu></yoo-vertical-menu>
```
```yoo-vertical-menu-modal.js 
  var menu = document.querySelector('yoo-vertical-menu');
  var verticalText = { text: "Amazing Component"};
  var verticalTextActive = { text: "Text", isActive: true };
  var verticalText2 = { text: "Core Component", imgSrc: "./assets/empty-states/calendar.svg" };
  var verticalText3 = { text: "Form Component", imgSrc: "./assets/empty-states/channel.svg", isActive: true };
  var verticalRow = { item: verticalText};
  var verticalRowActive = { item: verticalTextActive};
  var verticalRow2 = { item: verticalText2, subItems: [verticalText, verticalText3] };
  var verticalRow3 = { item: verticalText, subItems: [verticalText, verticalText, verticalText] };
  menu.entry = { menuRows: [verticalRowActive, verticalRow, verticalRow2, verticalRow, verticalRow3] };
  menu.fixed = false;
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`entry`|`VerticalMenuEntry`|{ menuRows: [] }|entry to create the vertical menu|
|`text`|`string`|'Menu'|text of the vertical menu|
|`fixed`|`boolean`|true|Display the vertical menu without a modal, as a fixed element|

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|yoo-vertical-menu accent|
|Dark|`dark`|yoo-vertical-menu dark|
|Success|`success`|yoo-vertical-menu success|
|Danger|`danger`|yoo-vertical-menu danger|
|Info|`info`|yoo-vertical-menu info|
|Warning|`warning`|yoo-vertical-menu warning|
|Gradient Accent|`gradient-accent`|yoo-vertical-menu gradient-accent|
|Gradient Dark|`gradient-dark`|yoo-vertical-menu gradient-grey|
|Gradient Success|`gradient-success`|yoo-vertical-menu gradient-success|
|Gradient Danger|`gradient-danger`|yoo-vertical-menu gradient-danger|
|Gradient Info|`gradient-info`|yoo-vertical-menu gradient-info|
|Gradient Warning|`gradient-warning`|yoo-vertical-menu gradient-warning|

## Events
|Attr|Description|
|---|---|
|`itemClicked`|event triggered when an item supplied to the menu has been clicked. Returned a string value defined in the entry prop as the emittedEvent string|
|`menuClosed`|event triggered when the menu is closed|
