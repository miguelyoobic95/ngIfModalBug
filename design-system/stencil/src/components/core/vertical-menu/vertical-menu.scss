:host {
  /deep/ div.modal-body,
  div.fixed-container {
    width: 100%;
    background-color: cssvar(darkblue);
    color: cssvar(light);
    .row {
      transition: all 0.5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
      a {
        color: cssvar(light);
        text-decoration: none;
        flex-grow: 1;
        &.active {
          //background-color: cssvar(dark-80);
          border-left: 0.25rem solid cssvar(accent);
          padding-left: 0.75rem;
        }
        &:hover {
          background-color: cssvar(darkblue-80);
        }
        .item {
          padding-left: 1rem;
          padding-top: 0.4rem;
          padding-bottom: 0.4rem;
          align-items: center;
          img {
            max-height: 1.5rem;
            max-width: 1.5rem;
            margin-right: 1rem;
          }
          i {
            margin-left: auto;
            margin-right: 1rem;
            transition: all 0.3s;
            &.chevron-active {
              transform: rotate(-90deg);
            }
          }
          &.active {
            background-color: cssvar(darkblue-80);
            border-left: 0.25rem solid cssvar(accent);
            padding-left: 0.75rem;
          }
          &:hover {
            background-color: cssvar(darkblue-80);
          }
        }
      }
      a:hover {
        text-decoration: none;
      }
      .sub-container {
        overflow: hidden;
        width: 100%;
        background: cssvar(darkblue);
        .row-subitems {
          overflow: hidden;
          animation: sub-item-appear 0.5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
          .item {
            padding-left: 2rem;
            &.active {
              padding-left: 1.75rem;
              background-color: cssvar(darkblue-60);
            }
            &:hover {
              background-color: cssvar(darkblue-60);
            }
          }
        }
      }
      &.hidden {
        .row-subitems {
          display: none;
        }
      }
      &.sub-display {
        background: cssvar(darkblue-80);
      }
    }
  }
  div.fixed-container {
    height: 100vh;
    div.menu-header {
      margin-left: 1rem;
      background: cssvar(darkblue);
      color: cssvar(light);
      span {
        font-size: 1.4rem;
      }
      .header-slot {
        padding: 1rem;
        .image {
          max-width: 90%;
        }
      }
    }
    .row {
      margin: 0px;
      .row-subitems {
        width: 100%;
      }
    }
  }
  /deep/ yoo-modal {
    .outer-container {
      background: cssvar(darkblue) !important;
      border: none !important;
      .modal-body {
        padding: 0 !important;
        padding-top: 5.0625rem !important;
      }
    }
  }
}

@keyframes sub-item-appear {
  0% {
    opacity: 0;
    transform: translateY(-100%);
  }
  100% {
    opacity: 1;
    transform: translateY(0%);
  }
}
