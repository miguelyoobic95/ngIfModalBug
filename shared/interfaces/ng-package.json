{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../node_modules/@shared/interfaces",
  "deleteDestPath": false,
  "lib": {
    "entryFile": "src/public_api.ts",
    "styleIncludePaths": [
      "../../../shared/styles"
    ],
    "umdModuleIds": {
      "moment": "moment",
      "moment-timezone": "momentTimezone",
      "lodash-es": "lodash-es"
    }
  }
}