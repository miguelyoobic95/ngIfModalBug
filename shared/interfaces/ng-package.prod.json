{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../../node_modules/@shared/interfaces",
  "lib": {
    "entryFile": "src/public_api.ts",
    "styleIncludePaths": [
      "../../../shared/styles"
    ]
  }
}