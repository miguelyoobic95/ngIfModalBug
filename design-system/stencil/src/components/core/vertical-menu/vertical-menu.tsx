import { Component, Prop, State, Event, EventEmitter, Element, Method } from '@stencil/core';
import { IVerticalMenuEntry, IVerticalMenuRow, IVerticalMenuItem } from '@shared/interfaces';
import { YooSlimScrollComponent } from '../slim-scroll/slim-scroll';

@Component({
    tag: 'yoo-vertical-menu',
    styleUrl: 'vertical-menu.scss',
    scoped: true
})
export class YooVerticalMenuComponent {

    @Prop({ mutable: true }) entry: IVerticalMenuEntry = { menuRows: [] };
    @Prop() fixed: boolean = true;
    @Prop() heading: string;
    @Prop() imgSrc: string;

    @Event() itemClicked: EventEmitter<IVerticalMenuItem>;
    @Event() menuClosed: EventEmitter<boolean>; // catch and re send close modal event

    @State() activeRow: boolean[] = [];

     @Element() host: HTMLStencilElement;

    onModalClosed() {
        this.menuClosed.emit(true);
    }

    componentWillLoad() {
        this.activeRow = this.entry.menuRows.map(() => false);
        if (this.fixed) {
            this.host.classList.add('fixed');
        }
    }

    componentDidLoad() {
        let slim: YooSlimScrollComponent = this.host.querySelector('yoo-slim-scroll') as any;
        if (slim) {
            slim.height = this.getInnerHeight();
            setTimeout(() => slim.refresh(), 200);
        }
    }

    componentDidUpdate() {
        let slim: YooSlimScrollComponent = this.host.querySelector('yoo-slim-scroll') as any;
        if (slim) {
            //slim.height = this.getInnerHeight();
            setTimeout(() => slim.refresh(), 200);
        }
    }

    getInnerHeight(): string {
        let header: HTMLElement = this.host.querySelector('.menu-header');
        if (header) {
            return (window.innerHeight - header.clientHeight) + 'px';
        }
        return '';
    }

    onItemClick(item: IVerticalMenuItem, index: number = null): void {
        this.itemClicked.emit(item);
        if (index || index === 0) {
            this.activeRow[index] = !this.activeRow[index];
            this.activeRow = this.activeRow.map((e) => e);
        }
        if (item) {
            this.setItemActive(item);
        }
    }

    @Method()
    setItemActive(menuItem: IVerticalMenuItem) {
        this.entry = {
            menuRows: this.entry.menuRows.map((row) => {
                row.item.isActive = row.item === menuItem && !row.item.isActive;
                if (row.subItems) {
                    row.subItems = row.subItems.map((item) => {
                        item.isActive = menuItem === item;
                        row.item.isActive = row.item.isActive || menuItem === item;
                        return item;
                    });
                }
                return row;
            })
        };
    }

    renderItem(item: IVerticalMenuItem, hasSubItem: boolean, index: number = null): JSX.Element {
        return (

            <a href={item.anchor ? item.anchor : null}>
                <div class={'item' + ((item.isActive) ? ' active' : '')}
                    onClick={() => this.onItemClick(item, index)}
                    attr-layout="row">
                    {item.imgSrc ? <img src={item.imgSrc} /> : null}
                    {item.icon ? <i class={item.icon} /> : null}
                    {item.text ?
                        <span>{item.text}</span>
                        : null}
                    {hasSubItem ?
                        <i class={'yo-arrow-dropdown' + ((this.activeRow[index]) ? ' chevron-active' : '')} />
                        : null}
                </div>
            </a>
        );
    }

    renderRow(row: IVerticalMenuRow, index): JSX.Element {
        if (row.subItems && row.subItems !== []) {
            return (
                <div class={'row' + ((this.activeRow[index]) ? ' sub-display' : ' hidden')}>
                    {this.renderItem(row.item, true, index)}
                    <div class="sub-container">
                        <div class={'row-subitems'}>
                            {row.subItems.map((subItem) =>
                                this.renderItem(subItem, false)
                            )}
                        </div>
                    </div>
                </div>
            );
        } else {
            return (
                <div class="row">
                    {this.renderItem(row.item, false)}
                </div>
            );
        }
    }

    render(): JSX.Element {
        return (
            this.fixed ?
                <div class="fixed-container" attr-layout="column">
                    <div class="menu-header">
                        <span>{this.heading}</span>
                        <div class="header-slot">
                            <img class="image" src={this.imgSrc} alt="Menu Header" />
                        </div>
                    </div>
                    <yoo-slim-scroll>
                        <div>
                            {this.entry.menuRows.map((row, index) =>
                                this.renderRow(row, index)
                            )}
                        </div>
                    </yoo-slim-scroll>
                </div>
                :
                <yoo-modal heading={this.heading ? this.heading : ''} class="menu dark" onClosed={() => this.onModalClosed()}>
                    <div attr-layout="column">
                        {this.entry.menuRows.map((row, index) =>
                            this.renderRow(row, index)
                        )}
                    </div>
                </yoo-modal>
        );
    }
}
