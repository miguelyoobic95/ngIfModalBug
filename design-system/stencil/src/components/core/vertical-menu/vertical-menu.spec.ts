import { TestWindow } from '@stencil/core/testing';
import { YooVerticalMenuComponent } from './vertical-menu';
import { IVerticalMenuRow, IVerticalMenuEntry, IVerticalMenuItem } from '@shared/interfaces';
import { YooModalComponent } from '../modal/modal';

describe('YooVerticalMenuComponent', () => {
    let window;
    beforeEach(() => {
        window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooVerticalMenuComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        it('Should render', async () => {
            let element = await window.load({
                components: [YooVerticalMenuComponent],
                html: '<yoo-vertical-menu></yoo-vertical-menu>'
            });
            expect(element).toBeTruthy();
        });

        it('Should render with only one Title', async () => {
            let element = await window.load({
                components: [YooVerticalMenuComponent],
                html: '<yoo-vertical-menu></yoo-vertical-menu>'
            });
            expect(element.entry).toEqual({ menuRows: [] });
            let verticalTitle: IVerticalMenuItem = { text: 'Title' };
            let verticalRow: IVerticalMenuRow = { item: verticalTitle };
            let entry: IVerticalMenuEntry = { menuRows: [verticalRow] };
            element.entry = entry;
            await window.flush();
            expect(element).toMatchSnapshot();
        });

        it('Should render with a prop title', async () => {
            let element = await window.load({
                components: [YooVerticalMenuComponent],
                html: '<yoo-vertical-menu title="My Menu"></yoo-vertical-menu>'
            });
            expect(element.entry).toEqual({ menuRows: [] });
            let verticalTitle: IVerticalMenuItem = { text: 'Title' };
            let verticalRow: IVerticalMenuRow = { item: verticalTitle };
            let entry: IVerticalMenuEntry = { menuRows: [verticalRow] };
            element.entry = entry;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with a fixed display', async () => {
            let element = await window.load({
                components: [YooVerticalMenuComponent],
                html: '<yoo-vertical-menu fixed="true"></yoo-vertical-menu>'
            });
            expect(element.entry).toEqual({ menuRows: [] });
            let verticalTitle: IVerticalMenuItem = { text: 'Title' };
            let verticalRow: IVerticalMenuRow = { item: verticalTitle };
            let entry: IVerticalMenuEntry = { menuRows: [verticalRow] };
            element.entry = entry;
            await window.flush();
            expect(element).toMatchSnapshot();
        });

        it('Should render correctly with a complex entry', async () => {
            let element = await window.load({
                components: [YooVerticalMenuComponent],
                html: '<yoo-vertical-menu></yoo-vertical-menu>'
            });
            expect(element.entry).toEqual({ menuRows: [] });
            let verticalTitle: IVerticalMenuItem = { text: 'Title' };
            let verticalTitleActive: IVerticalMenuItem = { text: 'Title', isActive: true };
            let verticalTitle2: IVerticalMenuItem = { text: 'Title Bis', imgSrc: './assets/grid/empty-state-1.svg' };
            let verticalTitle3: IVerticalMenuItem = { text: 'Title Ter', imgSrc: './assets/grid/empty-state-1.svg', isActive: true };
            let verticalRow: IVerticalMenuRow = { item: verticalTitle };
            let verticalRowActive: IVerticalMenuRow = { item: verticalTitleActive };
            let verticalRow2: IVerticalMenuRow = { item: verticalTitle2, subItems: [verticalTitle, verticalTitle3] };
            let verticalRow3: IVerticalMenuRow = { item: verticalTitle, subItems: [verticalTitle, verticalTitle, verticalTitle] };
            let entry: IVerticalMenuEntry = { menuRows: [verticalRowActive, verticalRow, verticalRow2, verticalRow, verticalRow3] };
            element.entry = entry;
            await window.flush();
            expect(element).toMatchSnapshot();
        });

    });

    // Event
    fdescribe('Events', () => {
        it('Should emit an event when closed', async () => {
            let element = await window.load({
                components: [YooVerticalMenuComponent, YooModalComponent],
                html: '<yoo-vertical-menu></yoo-vertical-menu>'
            });
            element.fixed = false;
            await window.flush();
            let modal = element.querySelector('yoo-modal');
            let res;
            let emitItem = jest.fn(ev => { res = ev.detail; });
            window.document.addEventListener('menuClosed', emitItem);
            await modal.close();
            expect(emitItem).toHaveBeenCalled();
            expect(res).toEqual(true);
        });

        it('Should emit an event when an item is clicked with the corresponding payload', async () => {
            let element = await window.load({
                components: [YooVerticalMenuComponent],
                html: '<yoo-vertical-menu></yoo-vertical-menu>'
            });
            expect(element.entry).toEqual({ menuRows: [] });
            let verticalTitle: IVerticalMenuItem = { text: 'Title' };
            let verticalTitleActive: IVerticalMenuItem = { text: 'Title', isActive: true };
            let verticalTitle2: IVerticalMenuItem = { text: 'Title Bis', imgSrc: './assets/grid/empty-state-1.svg', emittedEvent: 'title bis event' };
            let verticalTitle3: IVerticalMenuItem = { text: 'Title Ter', imgSrc: './assets/grid/empty-state-1.svg', isActive: true, emittedEvent: 'title ter event' };
            let verticalRow: IVerticalMenuRow = { item: verticalTitle };
            let verticalRowActive: IVerticalMenuRow = { item: verticalTitleActive };
            let verticalRow2: IVerticalMenuRow = { item: verticalTitle2, subItems: [verticalTitle, verticalTitle3] };
            let verticalRow3: IVerticalMenuRow = { item: verticalTitle, subItems: [verticalTitle, verticalTitle, verticalTitle] };
            let entry: IVerticalMenuEntry = { menuRows: [verticalRowActive, verticalRow, verticalRow2, verticalRow, verticalRow3] };
            element.entry = entry;
            await window.flush();

            let res;
            let emitItem = jest.fn(ev => { res = ev.detail; });
            window.document.addEventListener('itemClicked', emitItem);
            let items = element.querySelectorAll('.item');
            expect(items[0].classList).toContain('active');
            expect(items[1].classList).not.toContain('active');
            expect(items[2].classList).not.toContain('active');

            // First click
            await items[1].click();
            expect(emitItem).toHaveBeenCalled();
            expect(res).toEqual({ text: 'Title', isActive: true });
            //await window.flush();
            //expect(element.entry.menuRows[0].item.isActive).toEqual(true);

            // Second click with custom payload
            await items[2].click();
            expect(emitItem).toHaveBeenCalled();
            expect(res).toEqual(verticalTitle2);

            // Third click with custom payload
            await items[4].click();
            expect(emitItem).toHaveBeenCalled();
            expect(res).toEqual(verticalTitle3);
        });
    });
});
