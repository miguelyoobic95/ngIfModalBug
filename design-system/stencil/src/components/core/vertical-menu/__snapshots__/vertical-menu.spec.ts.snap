// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooVerticalMenuComponent Rendering Should render correctly with a complex entry 1`] = `
<yoo-vertical-menu
  class="fixed hydrated"
  data-yoo-vertical-menu-host=""
>
  <!---->
  <div
    attr-layout="column"
    class="fixed-container"
    data-yoo-vertical-menu=""
  >
    <div
      class="menu-header"
      data-yoo-vertical-menu=""
    >
      <span
        data-yoo-vertical-menu=""
      >
        Menu
      </span>
      <div
        class="header-slot"
        data-yoo-vertical-menu=""
        data-yoo-vertical-menu-slot=""
      />
    </div>
    <yoo-slim-scroll
      data-yoo-vertical-menu=""
    >
      <div
        data-yoo-vertical-menu=""
      >
        <div
          class="row"
          data-yoo-vertical-menu=""
        >
          <a
            data-yoo-vertical-menu=""
          >
            <div
              attr-layout="row"
              class="item active"
              data-yoo-vertical-menu=""
            >
              
              <span
                data-yoo-vertical-menu=""
              >
                Title
              </span>
              
            </div>
          </a>
        </div>
        <div
          class="row"
          data-yoo-vertical-menu=""
        >
          <a
            data-yoo-vertical-menu=""
          >
            <div
              attr-layout="row"
              class="item"
              data-yoo-vertical-menu=""
            >
              
              <span
                data-yoo-vertical-menu=""
              >
                Title
              </span>
              
            </div>
          </a>
        </div>
        <div
          class="row hidden"
          data-yoo-vertical-menu=""
        >
          <a
            data-yoo-vertical-menu=""
          >
            <div
              attr-layout="row"
              class="item"
              data-yoo-vertical-menu=""
            >
              <img
                data-yoo-vertical-menu=""
                src="./assets/grid/empty-state-1.svg"
              />
              
              <span
                data-yoo-vertical-menu=""
              >
                Title Bis
              </span>
              <i
                class="yo-arrow-dropdown"
                data-yoo-vertical-menu=""
              />
            </div>
          </a>
          <div
            class="sub-container"
            data-yoo-vertical-menu=""
          >
            <div
              class="row-subitems"
              data-yoo-vertical-menu=""
            >
              <a
                data-yoo-vertical-menu=""
              >
                <div
                  attr-layout="row"
                  class="item"
                  data-yoo-vertical-menu=""
                >
                  
                  <span
                    data-yoo-vertical-menu=""
                  >
                    Title
                  </span>
                  
                </div>
              </a>
              <a
                data-yoo-vertical-menu=""
              >
                <div
                  attr-layout="row"
                  class="item active"
                  data-yoo-vertical-menu=""
                >
                  <img
                    data-yoo-vertical-menu=""
                    src="./assets/grid/empty-state-1.svg"
                  />
                  
                  <span
                    data-yoo-vertical-menu=""
                  >
                    Title Ter
                  </span>
                  
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          class="row"
          data-yoo-vertical-menu=""
        >
          <a
            data-yoo-vertical-menu=""
          >
            <div
              attr-layout="row"
              class="item"
              data-yoo-vertical-menu=""
            >
              
              <span
                data-yoo-vertical-menu=""
              >
                Title
              </span>
              
            </div>
          </a>
        </div>
        <div
          class="row hidden"
          data-yoo-vertical-menu=""
        >
          <a
            data-yoo-vertical-menu=""
          >
            <div
              attr-layout="row"
              class="item"
              data-yoo-vertical-menu=""
            >
              
              <span
                data-yoo-vertical-menu=""
              >
                Title
              </span>
              <i
                class="yo-arrow-dropdown"
                data-yoo-vertical-menu=""
              />
            </div>
          </a>
          <div
            class="sub-container"
            data-yoo-vertical-menu=""
          >
            <div
              class="row-subitems"
              data-yoo-vertical-menu=""
            >
              <a
                data-yoo-vertical-menu=""
              >
                <div
                  attr-layout="row"
                  class="item"
                  data-yoo-vertical-menu=""
                >
                  
                  <span
                    data-yoo-vertical-menu=""
                  >
                    Title
                  </span>
                  
                </div>
              </a>
              <a
                data-yoo-vertical-menu=""
              >
                <div
                  attr-layout="row"
                  class="item"
                  data-yoo-vertical-menu=""
                >
                  
                  <span
                    data-yoo-vertical-menu=""
                  >
                    Title
                  </span>
                  
                </div>
              </a>
              <a
                data-yoo-vertical-menu=""
              >
                <div
                  attr-layout="row"
                  class="item"
                  data-yoo-vertical-menu=""
                >
                  
                  <span
                    data-yoo-vertical-menu=""
                  >
                    Title
                  </span>
                  
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </yoo-slim-scroll>
  </div>
</yoo-vertical-menu>
`;

exports[`YooVerticalMenuComponent Rendering Should render with a fixed display 1`] = `
<yoo-vertical-menu
  class="fixed hydrated"
  data-yoo-vertical-menu-host=""
  fixed="true"
>
  <!---->
  <div
    attr-layout="column"
    class="fixed-container"
    data-yoo-vertical-menu=""
  >
    <div
      class="menu-header"
      data-yoo-vertical-menu=""
    >
      <span
        data-yoo-vertical-menu=""
      >
        Menu
      </span>
      <div
        class="header-slot"
        data-yoo-vertical-menu=""
        data-yoo-vertical-menu-slot=""
      />
    </div>
    <yoo-slim-scroll
      data-yoo-vertical-menu=""
    >
      <div
        data-yoo-vertical-menu=""
      >
        <div
          class="row"
          data-yoo-vertical-menu=""
        >
          <a
            data-yoo-vertical-menu=""
          >
            <div
              attr-layout="row"
              class="item"
              data-yoo-vertical-menu=""
            >
              
              <span
                data-yoo-vertical-menu=""
              >
                Title
              </span>
              
            </div>
          </a>
        </div>
      </div>
    </yoo-slim-scroll>
  </div>
</yoo-vertical-menu>
`;

exports[`YooVerticalMenuComponent Rendering Should render with a prop title 1`] = `
<yoo-vertical-menu
  class="fixed hydrated"
  data-yoo-vertical-menu-host=""
  title="My Menu"
>
  <!---->
  <div
    attr-layout="column"
    class="fixed-container"
    data-yoo-vertical-menu=""
  >
    <div
      class="menu-header"
      data-yoo-vertical-menu=""
    >
      <span
        data-yoo-vertical-menu=""
      >
        Menu
      </span>
      <div
        class="header-slot"
        data-yoo-vertical-menu=""
        data-yoo-vertical-menu-slot=""
      />
    </div>
    <yoo-slim-scroll
      data-yoo-vertical-menu=""
    >
      <div
        data-yoo-vertical-menu=""
      >
        <div
          class="row"
          data-yoo-vertical-menu=""
        >
          <a
            data-yoo-vertical-menu=""
          >
            <div
              attr-layout="row"
              class="item"
              data-yoo-vertical-menu=""
            >
              
              <span
                data-yoo-vertical-menu=""
              >
                Title
              </span>
              
            </div>
          </a>
        </div>
      </div>
    </yoo-slim-scroll>
  </div>
</yoo-vertical-menu>
`;

exports[`YooVerticalMenuComponent Rendering Should render with only one Title 1`] = `
<yoo-vertical-menu
  class="fixed hydrated"
  data-yoo-vertical-menu-host=""
>
  <!---->
  <div
    attr-layout="column"
    class="fixed-container"
    data-yoo-vertical-menu=""
  >
    <div
      class="menu-header"
      data-yoo-vertical-menu=""
    >
      <span
        data-yoo-vertical-menu=""
      >
        Menu
      </span>
      <div
        class="header-slot"
        data-yoo-vertical-menu=""
        data-yoo-vertical-menu-slot=""
      />
    </div>
    <yoo-slim-scroll
      data-yoo-vertical-menu=""
    >
      <div
        data-yoo-vertical-menu=""
      >
        <div
          class="row"
          data-yoo-vertical-menu=""
        >
          <a
            data-yoo-vertical-menu=""
          >
            <div
              attr-layout="row"
              class="item"
              data-yoo-vertical-menu=""
            >
              
              <span
                data-yoo-vertical-menu=""
              >
                Title
              </span>
              
            </div>
          </a>
        </div>
      </div>
    </yoo-slim-scroll>
  </div>
</yoo-vertical-menu>
`;
